storage/sql/template.go

```diff
@@ -145,34 +199,167 @@ func (rs *RecordStream) Close() error {
 	return nil
 }
 
-func (rs *RecordStream) Next(ctx context.Context) (*execution.Record, error) {
+func (rs *RecordStream) RunWorker(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			rs.workerCloseErrChan <- ctx.Err()
+		default:
+		}
+
+		tx := rs.stateStorage.BeginTransaction().WithPrefix(rs.streamID.AsPrefix())
+
+		err := rs.RunWorkerInternal(ctx, tx)
+		if errors.Cause(err) == execution.ErrNewTransactionRequired {
+			tx.Abort()
+			continue
+		} else if waitableError := execution.GetErrWaitForChanges(err); waitableError != nil {
+			tx.Abort()
+			err = waitableError.ListenForChanges(ctx)
+			if err != nil {
+				log.Println("sql worker: couldn't listen for changes: ", err)
+			}
+			err = waitableError.Close()
+			if err != nil {
+				log.Println("sql worker: couldn't close storage changes subscription: ", err)
+			}
+			continue
+		} else if err != nil {
+			tx.Abort()
+			log.Printf("sql worker: error running sql read batch worker: %s, reinitializing from storage", err)
+			tx := rs.stateStorage.BeginTransaction().WithPrefix(rs.streamID.AsPrefix())
+			if err := rs.loadOffset(tx); err != nil {
```

Comment:
We rarely do `if err := someFunction(); err != nil`. I think it's better to split in into the standard two lines:
`err := someFunction()`
`if err != nil ...`

Reply:
This is exactly how it is written in kafka, I was just sticking to it. But yea, can fix that :)